exonum-java-binding/doc/Migration_guide_0.9.md

```diff
@@ -0,0 +1,183 @@
+
+## Dynamic Services Overview
+
+Dynamic Java services are packaged in an _artifact_: a JAR archive with some metadata. An Exonum
+artifact has a runtime-specific _name_. Java services must use name in the format 
+'groupId:artifactId:version', where each component of the name corresponds
+to a Maven project coordinate, for example: 'com.exonum.example:timestamping:1.0.2' is a valid
+artifact name. Rust service names are usually in format '<crate-name>:<version>'.
+The format of the metadata has not changed since 0.6.0.
+
+Artifacts are _deployed_ to the Exonum blockchain network; and then used to instantiate
+_service instances_. A single artifact can be used to create many instances. Each instance
+has a _name_ (ex `Service#getName`) and a numeric _identifier_ (ex `Service#getId`).
+An instance name is specified by the network administrators during service instantiation;
+an id is assigned by the framework automatically and can be [queried][service-id-lc-operation]
+using the light client.
+
+[service-id-lc-operation]: todo
+
+## Update the Service
+
+The [`Service`][service] implementation must be updated to support multiple instances;
+new initialization and reconfiguration procedures.
+
+1. Remove the hardcoded service id and name. Add a constructor parameter 
+`ServiceInstanceSpec instanceSpec` which provides instance name
+and id.
+    - If your service extends
+    [`com.exonum.binding.core.service.AbstractService`][abstract-service] —
+    pass the `instanceSpec` to the superclass constructor and use provided `AbstractService#getId`,
+    `getName`, and `getInstanceSpec` methods
+    - If it implements `Service` directly — save the needed information to fields.
+
+```java
+public final class FooService extends AbstractService {
+  @Inject
+  public FooService(ServiceInstanceSpec instanceSpec) {
+    super(instanceSpec);
+  }
+}
+```
+
+2. If the service requires initialization — migrate to the new 
+[initialization mechanism][service-initialize]. 
+
+3. If the service needs reconfiguration support: implement [`Configurable`][configurable] interface.
+Reconfiguration shall be performed through the new [supervisor][supervisor] service which
+replaces the configuration service.
+
+[service]: https://exonum.com/doc/api/java-binding/0.9.0-rc1/com/exonum/binding/core/service/Service.html
+[service-initialize]: https://exonum.com/doc/api/java-binding/0.9.0-rc1/com/exonum/binding/core/service/Service.html#initialize-com.exonum.binding.core.storage.database.Fork-com.exonum.binding.core.service.Configuration-
+[configurable]: https://exonum.com/doc/api/java-binding/0.9.0-rc1/com/exonum/binding/core/service/Configurable.html
+<!-- todo: Check the anchor when the docs land -->
+[supervisor]: https://exonum.com/doc/version/0.13-rc1/get-started/java-binding/#deploy-and-start-service
+
+### Update the Schema
+
+#### Isolate Collections via Namespaces
+
+Use a unique for each service instance namespace for service collections (e.g., a service name,
```

Comment:
```suggestion
Use a unique instance namespace for each service in service collections (e.g., a service name,
```

Reply:
We use 'service instance' as a term to highlight, when needed, the difference between _service artifacts_ and _service instances_. unique — among all service instances (even of all types).

With this patch it might also appear that there are services in collections.

Reply:
Maybe, "Use a namespace for/in the service collections that is unique among all service instances (e.g., ....)"